### ISSUE TITLE
External content included in dragged list item (Safari only)

### ISSUE THREAD

**Issue by hcampos-professional** [Esri/calcite-design-system#9467]
**monday.com sync:** #18048571908

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

https://github.com/Esri/calcite-design-system/assets/136828178/ca6fe749-610f-4886-832e-8537fcf38e7d

When dragging a list item which contains a `calcite-action-menu`, other content that is not inside of the list item is included in the "ghost" image, but only in Safari. It seems like Safari includes anything within the "bounds" of the dragged element and, because of the action menu being larger, extra content gets included, even if it is not in the list item.

### Expected Behavior

The expected behavior is that only elements that are inside of the list item get included in the ghost image. In Chrome/Edge, the behavior is as expected.

### Reproduction Sample

https://codesandbox.io/p/sandbox/calcite-nested-list-flickering-forked-zlkpv8?file=%2Findex.html%3A7%2C1

### Reproduction Steps

1. Open https://codepen.io/hccampos/pen/yLWgKJp in Safari
2. Drag the single list item. Notice that the "OUTSIDE CONTENT" text gets included in the ghost while dragging.

### Reproduction Version

2.8.4

### Relevant Info

Only happens in Safari (tested with Version 17.5 (19618.2.12.11.6))

### Priority impact

impact - p3 - not time sensitive

### Impact

It impacts the `LayerList` widget of the ArcGIS Maps SDK, which is used in the ArcGIS SceneViewer and ArcGIS MapViewer, for example.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by driskull** [comment]
Seems to only be happening when the action-menu/group is present.

**Comment by hcampos-professional** [comment]
Yes, and the content is only included until the bottom of the `action-menu`. It really seems like Safari just takes a snapshot of the bounds of the element, without considering whether an element is even visible or part of the element being dragged.

**Comment by driskull** [comment]
Yeah, i'll see if there's a workaround otherwise I might have to find a safari bug to link this to or open one.

**Comment by jcfranco** [comment]
The recent updates to draggable list items make this less noticeable, but it’s still reproducible if you move an item while the bottom `action-menu` is open. **Note**: This now requires at least two items, as dragging isn’t enabled otherwise.

**Comment by isaacbraun** [comment]
Still present and was brought up during the verification of [12204](https://github.com/Esri/calcite-design-system/issues/12204#issuecomment-3152921164). Requires the drag menu to be open and still only presenting in Safari. Matt's [most recent comment](https://github.com/Esri/calcite-design-system/issues/12204#issuecomment-3155716428) suggested that this is a Safari issue.

<img width="2560" height="1440" alt="Image" src="https://github.com/user-attachments/assets/c8f1f28a-38dd-409e-9774-2fa5ad3d41e5" />

**Comment by isaacbraun** [comment]
After more digging, it seems this is an issue others have run into with Sortable and Safari. It seems to be related to Safari's implementation of HTML5's Drag 'n Drop, in that it incorrectly calculates the height or area of elements that it should "ghost". Even so, there may be some options we can implement to mitigate it. 

I've created a non-Calcite reproduction CodePen: https://codepen.io/isaacbraun/pen/JoYLXEG
1. In Safari, open one of the dropdown buttons and drag the element.
2. Observe that the other element's text will be included in the ghost.
3. Also note, that the `a` element text will be included in the ghost for the bottom item, but if the display is set to `inline-block` or similar it does not get included. Additionally, a `button` element does not get included, likely due to the same `display` property reason.

There are a few issues found in Sortable's repo that discuss this behavior, including some potential solutions.
- [discussion of cause](https://github.com/SortableJS/Sortable/issues/291): Safari chooses the height of what to ghost based on the height of the element or child elements, it our case the open drag menu dropdown of `calcite-list-item`.
- [user-select: none](https://github.com/SortableJS/Sortable/issues/997): this can potentially work, but the rule would have to applied to any element that needs to be "ignored" by the drag ghost.
- [forceFallback](https://github.com/SortableJS/Sortable/issues/1841): this seems to work. It disables the HTML5 Drag 'n Drop, and can be conditionally set based on the browser type, so only Safari would utilize the fallback.

I am considering this Spike completed, but I'm happy to dig more if needed.

cc @jcfranco @driskull

**Comment by driskull** [comment]
Nice work @isaacbraun. Were you able to find a Safari issue we can link to? I'm sure one exists for this and if not we should create one.

Would `user-select:none` work on the drag/ghost class possibly? Otherwise, I don't think the other workarounds would be sufficient.

**Comment by isaacbraun** [comment]
Thanks Matt 😄 

I have not found a direct match for a Safari issue on https://bugs.webkit.org. If I should be looking elsewhere, please let me know. I'm happy to open an issue if needed, but I did find a few that are related:
- [Drag image incorrect size when child out of bounds](https://bugs.webkit.org/show_bug.cgi?id=227630): this is the most relevant
- Drag image not shown when element or parent element is set to `display: inline-block`:
	- https://bugs.webkit.org/show_bug.cgi?id=35754
	- https://bugs.webkit.org/show_bug.cgi?id=88745
	- https://bugs.webkit.org/show_bug.cgi?id=189424
- [Drag image does not respect CSS transition](https://bugs.webkit.org/show_bug.cgi?id=250721)

RE: `user-select: none`, I don't think so. I've updated the [CodePen](https://codepen.io/isaacbraun/pen/JoYLXEG) to use a ghost class with those rules applied and nothing seems to change. I think the selection would have to be disabled on every element we _wouldn't_ want selected in the drag image (ghost).

What are the scenarios/reasons the the `forceFallback` solution does not work? If you uncomment that line in the Sortable initialization it seems to work as expected.

**Comment by isaacbraun** [comment]
That makes complete sense, appreciate the explanation 👍  Will get a Safari bug submitted.

**Comment by isaacbraun** [comment]
WebKit team has begun work: https://github.com/WebKit/WebKit/pull/50105
